pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -426,9 +426,28 @@ func (c *Cluster) startDeployments(mons []*monConfig, requireAllInQuorum bool) e
 		if err != nil {
 			return fmt.Errorf("failed to create mon %s. %+v", mons[i].DaemonName, err)
 		}
+		// For the initial deployment (first creation) it's expected to not have all the monitors in quorum
+		// However, in an event of an update, it's crucial to proceed monitors by monitors
+		// At the end of the method we perform one last check where all the monitors must be in quorum
+		requireAllInQuorum := false
+		err = c.waitForMonsToJoin(mons, requireAllInQuorum)
```

Comment:
Per our conversation, if the ok-to-stop check for mons is sufficient, we don't need the changes in this method to wait for quorum. If we change the logging to debug for some of those messages, would that be sufficient for the logging concern? This extra wait for quorum would still be good to avoid. Our mon handling is already complex enough, I don't want to add it if it's not necessary.

Reply:
The problem with this is that I can't possibly implement this in the client package since I'll need to:

* export `waitForMonsToJoin` which is easy
* then import `mon "github.com/rook/rook/pkg/operator/ceph/cluster/mon"` in the client package

Then I'll end up with a circular dependency... Same goes for clusterInfo, coming from cephconfig.

I know the mon code is complex but I've put enough comments to clarify when this code is getting triggered.

If you have another suggestion, please let me know.